actor/divine_smite.js

```diff
@@ -7,94 +7,131 @@
  * attack and then if you choose to use Divine Smite, run this macro.
  */
 
-let confirmed = false;
-
-// Create a dialogue box to select spell slot level to use when smiting.
-new Dialog({
-    title: "Divine Smite Damage",
-    content: `
-     <p>Spell Slot level to use Divine Smite with.</p>
-     <form>
-      <div class="form-group">
-       <label>Spell Slot Level:</label>
-       <select id="slot-level" name="slot-level">
-        <option value="1">1</option>
-        <option value="2">2</option>
-        <option value="3">3</option>
-        <option value="4">4</option>
-        <option value="5">5</option>
-       </select>
+ //Configurable variables
+ let maxSpellSlot = 5; //Highest spell-slot level that may be used.
+ let affectedCreatureTypes = ["fiend", "undead", "undead (shapechanger)"]; //Creature types that take extra damage.
+ let allowWithoutSpellSlot = false; //Allows the macro to be used without expending a spell slot. 
+ let macroFlavor = `Macro Divine Smite - Damage Roll (Radiant)`; //Flavor to show in the chat roll.
+ let dieSize = `d8`; //Die size to use for the SMITE.
+ //
+ 
+ let optionsText = `<p>Spell Slot level to use Divine Smite with.</p><form>
+ <div class="form-group"><label>Spell Slot Level:</label><select id="slot-level" name="slot-level">`
+ 
+ let s_actor = canvas.tokens.controlled[0].actor;
+ 
+ if (s_actor === undefined){
+    s_actor = game.user.character;     
+ }
```

Comment:
Should be

`let s_actor = canvas.tokens.controlled[0]?.actor || game.user.character;`

Otherwise it will launch an error when no token is controlled.